plugins/ipstack/icon_ipstack/connection/connection.py

```diff
@@ -13,4 +16,18 @@ def connect(self, params):
         self.token = params.get(Input.CRED_TOKEN).get("secretKey")
 
     def test(self):
-        return {"success": True}
+        # Sending a request using the secretKey to test if it is valid
+        url = "http://api.ipstack.com/" + "rapid7.com" + "?access_key=" + self.token + "&output=json"
+
+        invalid_status_codes = ["404", "101", "102", "104", "106"]
+
+        resp = insightconnect_plugin_runtime.helper.open_url(url)
+        dic = json.loads(resp.read())
+        if any(invalid_status_codes) in dic or "error" in dic:
+            raise PluginException(
+                data=dic,
+                assistance="Please check your credentials and try again. If the issue persists, please contact support.",
+                cause="An error has occurred. Please check your credentials are correct and that you have not exceeded the lookup limits.",
+            )
+        else:
+            return {"success": True}
```

Comment:
I suggest using `example.com` instead of `rapid7.com`.

```suggestion
        url = f"http://api.ipstack.com/example.com?access_key={self.token}&output=json"
        response = insightconnect_plugin_runtime.helper.open_url(url)
        parsed_response = json.loads(response.read())
        invalid_status_codes = ["404", "101", "102", "104", "106"]
        if any(invalid_status_codes) in parsed_response or "error" in dic:
            raise PluginException(
                data=parsed_response,
                assistance="Please check your credentials and try again. If the issue persists, please contact support.",
                cause="An error has occurred. Please check your credentials are correct and that you have not exceeded the lookup limits.",
            )
        return {"success": True}
```